src/color.js

```diff
@@ -0,0 +1,461 @@
+"use strict";
+
+/**
+ * color module
+ * @module color
+ */
+
+/**
+ * Parses a color string in rgb or rgba format into a Color object.
+ *
+ * @param {string} colorString - The color string from CSS.
+ * @return {Color}
+ */
+export function parseColor (colorString) {
+  if (colorString === "transparent") {
+    return new Color(0, 0, 0, 0);
+  }
+
+  let rgbRegex = /^rgb\((\d+), (\d+), (\d+)\)$/;
+  let match = colorString.match(rgbRegex);
+
+  if (match) {
+    let r = parseInt(match[1], 10);
+    let g = parseInt(match[2], 10);
+    let b = parseInt(match[3], 10);
+    let a = 1;
+
+    return new Color(r, g, b, a);
+  }
+
+  let rgbaRegex = /^rgba\((\d+), (\d+), (\d+), (\d*(\.\d+)?)\)/;
+  match = colorString.match(rgbaRegex);
+
+  if (match) {
+    let r = parseInt(match[1], 10);
+    let g = parseInt(match[2], 10);
+    let b = parseInt(match[3], 10);
+    let a = parseFloat(match[4]);
+
+    return new Color(r, g, b, a);
+  }
+
+  return null;
+};
+
+/**
+ * Suggests alternative color suggestions to meet a given contrast ratio.
+ *
+ * @param {Color} bgColor - the background color.
+ * @param {Color} fgColor - the foreground color.
+ * @param {Object.<string, number>} desiredContrastRatios - A map of label to desired contrast ratio.
+ * @return {Object.<string, string>}
+ */
+export function suggestColors (bgColor, fgColor, desiredContrastRatios) {
```

Comment:
Or, alternatively, we could accept `{ red, green, blue, alpha }` objects here - my comment should have been on `parseColor` above, which maybe should be private?